server/memcached/src/main/java/org/infinispan/server/memcached/binary/BinaryIntrinsics.java

```diff
@@ -30,7 +31,10 @@ public static short short_(ByteBuf b) {
       } else return 0;
    }
 
-   public static byte[] fixedArray(ByteBuf b, int length) {
+   public static byte[] fixedArray(ByteBuf b, int length, int maxArrayLength) {
+      if (maxArrayLength > 0 && length > maxArrayLength) {
+         throw new TooLongFrameException("Array length " + length + " exceeded " + maxArrayLength);
```

Comment:
This code seems to be duplicated all over the place (only slightly changing the message). Would it make sense to move it to `server/core` in some util class?

Reply:
I kept debating about this, I guess we can. I think I can reduce it down to 4 variants.